lib/ontologies_linked_data/models/mod/semantic_artefact_catalog_record.rb

```diff
@@ -0,0 +1,112 @@
+      attribute_mapped :curatedOn, namespace: :pav, mapped_to: { model: :ontology_submission }
+      attribute_mapped :curatedBy, namespace: :pav, mapped_to: { model: :ontology_submission }
+
+      # additional attributes
+      attribute_mapped :viewingRestriction, mapped_to: { model: :ontology }
+      attribute_mapped :administeredBy, mapped_to: { model: :ontology }
+      attribute_mapped :doNotUpdate, mapped_to: { model: :ontology }
+      attribute_mapped :flat, mapped_to: { model: :ontology }
+      attribute_mapped :summaryOnly, mapped_to: { model: :ontology }
+      attribute_mapped :acl, mapped_to: { model: :ontology }
+      attribute_mapped :ontologyType, mapped_to: { model: :ontology }
+      attribute_mapped :classType, mapped_to: { model: :ontology_submission }
+      attribute_mapped :missingImports, mapped_to: { model: :ontology_submission }
+      attribute_mapped :submissionStatus, mapped_to: { model: :ontology_submission }
+      attribute_mapped :pullLocation, mapped_to: { model: :ontology_submission }
+      attribute_mapped :dataDump, namespace: :void, mapped_to: { model: :ontology_submission }
+      attribute_mapped :csvDump, mapped_to: { model: :ontology_submission }
+      attribute_mapped :uploadFilePath, mapped_to: { model: :ontology_submission }
+      attribute_mapped :diffFilePath, mapped_to: { model: :ontology_submission }
+      attribute_mapped :masterFileName, mapped_to: { model: :ontology_submission }
+
+
+      attribute :ontology, type: :ontology, enforce: [:existence]
+      attribute :submission, type: :ontology_submission
+
+      # Access control
+      read_restriction_based_on ->(record) { record.ontology }
+
+      serialize_default :acronym, :relatedArtefactId, :homepage, :created, :modified, :curatedOn, :curatedBy
+      serialize_never :ontology, :submission
+
+      def self.record_id_generator(record)
+        record.ontology.bring(:acronym) if !record.ontology.loaded_attributes.include?(:acronym)
+        raise ArgumentError, "Acronym is nil for ontology  #{record.ontology.id} to generate id" if record.ontology.acronym.nil?
+        return RDF::URI.new(
+          "#{(Goo.id_prefix)}records/#{CGI.escape(record.ontology.acronym.to_s)}"
+        )
+      end
+
+      ##
+      ## find an artefact (ontology) and map it to record
+      def self.find(artefact_id)
+        ont = Ontology.find(artefact_id).include(:acronym, :viewingRestriction, :administeredBy, :acl).first
+        return nil unless ont
+        new.tap do |sacr|
+          sacr.ontology = ont
+        end
+      end
+      
+      def self.all(attributes, page, pagesize)
+        all_count = Ontology.where.count
+        onts = Ontology.where.include(:acronym, :viewingRestriction, :administeredBy, :acl).page(page, pagesize).page_count_set(all_count).all
+        all_records = onts.map do |o|
+          new.tap do |sacr|
+            sacr.ontology = o
+            sacr.bring(*attributes) if attributes
```

Comment:
the bring attributes in the line 66

Reply:
otherwise will be slow

Reply:
in this we cannot include the attirbutes in the line 66 because they are mixed from different models. Also i don't think it will be that slow because it uses the bring method in `attribute_fetcher.rb` and i fixed that method to group first the attribute and bring them in one shot (same logic as SemanticArtefacts)